Sources/SKTestSupport/SkipUnless.swift

```diff
@@ -0,0 +1,252 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LSPLogging
+import LanguageServerProtocol
+import RegexBuilder
+@_spi(Testing) import SKCore
+import XCTest
+
+import enum PackageLoading.Platform
+import struct TSCBasic.AbsolutePath
+import class TSCBasic.Process
+import enum TSCBasic.ProcessEnv
+
+// MARK: - Skip checks
+
+/// Namespace for functions that are used to skip unsupported tests.
+public enum SkipUnless {
+  private enum FeatureCheckResult {
+    case featureSupported
+    case featureUnsupported(skipMessage: String)
+  }
+
+  /// For any feature that has already been evaluated, the result of wheather or not it should be skipped.
+  private static var checkCache: [String: FeatureCheckResult] = [:]
+
+  /// Throw an `XCTSkip` if any of the following conditions hold
+  ///  - The Swift version of the toolchain used for testing (`ToolchainRegistry.forTesting.default`) is older than
+  ///    `swiftVersion`
+  ///  - The Swift version of the toolchain used for testing is equal to `swiftVersion` and `featureCheck` returns
+  ///    `false`. This is used for features that are introduced in `swiftVersion` but are not present in all toolchain
+  ///    snapshots.
+  ///
+  /// Having the version check indicates when the check tests can be removed (namely when the minimum required version
+  /// to test sourcekit-lsp is above `swiftVersion`) and it ensures that tests can’t stay in the skipped state over
+  /// multiple releases.
```

Comment:
Heh, thanks for the comment. I wrote a comment asking what the version is for :).

Reply:
> Heh, thanks for the comment. I wrote a comment asking what the version is for then had to delete it after reading more :).